### DISCUSSION TITLE
Infinite live location

### DISCUSSION THREAD

**Discussion by PeterHunt** [windkh/node-red-contrib-telegrambot#370] [General]
Dear Karl-Heinz,

I just noticed that the latest Telegram update contains the possibility for infinite live location, until stopped manually.
I'm not really sure if your BOT is showing all details? When sending a fixed location, the live_period is not present. When sending a live location for a certain period, the live_period *is* present.

But, when setting to infinite (just released, I updated 2 different Android phones today), the live_period is also not present. Which makes a bit of sense, since it will never end, or at least not automatically.

My question is now KH, can you verify that there is no additional data present in this new option?
So the payload for fixed and infinite live location is exactly the same?

If so, I might face a challenge to implement this :)

Hoping to hear from you.

Kind regards,

Peter Hunt.

**Comment by windkh** [comment]
Hi Peter,
to be honest: I did not hear about that feature until now. If it is part of an API update, then an update of the underlying library to the latest version is required. 
I will check this as soon as I find some minutes...
Br kh

**Comment by windkh** [comment]
Did you check the property originalMessage inside msg?

**Comment by PeterHunt** [comment]
Yes I did, this is what I see :

object
payload: object
originalMessage: object
message_id: 57593239
from: object
chat: object
date: 1714166487
location: object
latitude: 53.059624
longitude: 5.469624
_msgid: "b5d86993feb2594e

**Reply by windkh** [reply]
Please expand the originalMessage node and check if there is a location property ...

**Comment by PeterHunt** [comment]
When sending the live location, I see :

msg : Object
originalMessage: object
message_id: 57743500
from: object
chat: object
date: 1714248216
**location: object
latitude: 53.059649
longitude: 5.469627**

When the location gets updated (Edited message) :

msg : Object
originalMessage: object
message_id: 57743500
from: object
chat: object
date: 1714248216
edit_date: 1714248252
**location: object
latitude: 53.059652
longitude: 5.469621
horizontal_accuracy: 3**

**Comment by PeterHunt** [comment]
Hello,

Is there any update maybe, or am I overseeing something?

BR, Peter Hunt

**Comment by windkh** [comment]
Oh sorry... due to too many things in my head I forgot about that... as soon as I find some time I will check it out

**Comment by windkh** [comment]
@PeterHunt 
I did some tests with the following example
https://github.com/windkh/node-red-contrib-telegrambot/blob/master/examples/livelocation.json

1. I modified the /send command that sends the initial live location:
```
msg.payload.options = {
    live_period: 0x7FFFFFFF
};  
```
--> this shows the location of the bot on my iPhone: works as expected

2. I modified the /update command (editMessageLiveLocation)
```
msg.payload.options = {
    chat_id : chatId,
    message_id : messageId,
    live_period: 0x7FFFFFFF
}; 
```
--> this also works as expected. 

3. Then I did it the other way around: I activated sending my iPhones live location so that the bot received the location messages (here I chose "Until I turn it off" from the options)
--> the location receiver node received the following

```
location: object
latitude: 47.933075
longitude: 10.719416
live_period: 2147483647
heading: 339
horizontal_accuracy: 4
```

So everything works as expected. Have I misunderstood something?

**Comment by PeterHunt** [comment]
Karl-Heinz,

At this point, I don't understand no more, I will try and explain why.

To start with the good news; you are right. NOW it is working.

I say now, because when I reported this I was testing myself, after a notification of one of my users. There was definitely no live_period present when I reported this issue. This was confirmed by the BOTs behaviour, since it constantly registered a new fixed location (not live location). This was reproduced over and over again.

Maybe Telegram added something just after my report, to be honest I didn't performed any more testing...

The only suggestion I have is that Telegram launched something which was not quite finished?

Anyways, thanks for your time (as allways), it is appreciated! :)
